plugin/server/datastore-sqlite/sqlite.go

```diff
@@ -27,12 +28,16 @@ var (
 )
 
 type sqlitePlugin struct {
-	db *gorm.DB
+	db    *gorm.DB
+	mutex *sync.Mutex
```

Comment:
shouldn't sqlite be handling record locking? Lots of the operations here can be performed in parallel

Reply:
There might be a better way of handling locking using sqlite itself. I can create an issue to look into this.